packages/query/src/block-processor.ts

```diff
@@ -125,7 +136,7 @@ export class BlockProcessor implements BlockProcessorInterface {
     this.numeraires = numeraires;
   }
 
-  private async syncAndStore() {
+  private async syncAndStore(genesisBlock?: CompactBlock) {
```

Comment:
should we make type signature here more explicit, `genesisBlock: CompactBlock | undefined`?

Reply:
Given it is a class field we don't need to pass it here. We can just reference it below:

```diff
- if (currentHeight === -1n && genesisBlock) {
+ if (currentHeight === -1n && this.genesisBlock) {
```